lib/spack/docs/packaging_guide.rst

```diff
@@ -2521,6 +2521,100 @@ would do with the package's own spec:
 
    mpicc = join_path(my_mpi.prefix.bin, 'mpicc')
 
+^^^^^^^^^^^^^^^^^^^^
+Spec build interface
+^^^^^^^^^^^^^^^^^^^^
+
+For more complex queries Spack supports an extended sub-scripting syntax:
+
+.. code-block:: python
+
+   mpi_cxx_and_fortran_query = spec['mpi:cxx,fortran']
+
+which allows for a list of comma separated values to be specified after
+the package name (a single colon separates the package name from the values).
+These values will be attached to the ``Spec`` object being created and can be
+used by the dependency to fine tune the computation of certain attributes:
+
```

Comment:
A `Spec` instance which additionally knows that somebody asked for `mpi` with `cxx` and `fortran` as extra query parameters.